### ISSUE TITLE
Getting the predicted question type from `type_prediction.py`

### ISSUE THREAD

**Issue by carriex** [ShuyangCao/open-ended_question_ontology#2]
Hi there,

Thanks for sharing the code and data! I am trying to run the question type classifier in `prediction_scripts/type_prediction.py`. I was able to get a list of predicted indexes from running the script with the pretrained model. I am wondering what's the mapping between the label index to the actual type name (e.g. "Verification", "Disjunctive" etc.)

thanks!

**Comment by ShuyangCao** [comment]
Hi,

Here's the mapping:

1: Verification
2: Disjunctive
3: Concept
5: Extent
7: Example
8: Comparison
9: Judgmental
10: Cause
11: Consequence
13: Procedural

**Comment by carriex** [comment]
thanks!

**Comment by carriex** [comment]
Hi @ShuyangCao , 

I tried to run the prediction script (with question input downloaded from [here](https://drive.google.com/drive/folders/1LIccouKa7yfv4axyt_XAh56d5XR3WuDv)) and the output based on this mapping looks a bit weird. Specifically, when I run prediction on the 4,959 annotated questions (downloaded from [here](https://drive.google.com/drive/folders/1eipP3o5d_8aGqks3EaOTJxxgYe_oD3vQ)), I only got a macro F1 of 0.19 compared to the resolved type -- which is unexpected given that these data contains training data for the prediction model. 

The confusion matrix looks like this: 

type | concept | verification | procedural | comparison | cause | judgmental | extent | example | disjunction | consequence
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
**concept** | 2 | 299 | 17 | 3 | 10 | 39 | 2 | 15 | 2 | 0
**verification** | 431 | 6 | 62 | 1 | 3 | 47 | 8 | 15 | 2 | 0
**procedural** | 2 | 5 | 4 | 1 | 8 | 570 | 0 | 26 | 1 | 0
**comparison** | 2 | 35 | 16 | 0 | 3 | 9 | 107 | 45 | 0 | 0
**cause** | 2 | 16 | 6 | 0 | 585 | 22 | 0 | 6 | 0 | 0
**judgmental** | 25 | 25 | 431 | 1 | 12 | 22 | 25 | 103 | 0 | 0
**extent** | 13 | 61 | 80 | 167 | 0 | 21 | 16 | 11 | 6 | 0
**example** | 12 | 28 | 38 | 0 | 1 | 35 | 1 | 1047 | 4 | 0
**disjunction** | 65 | 2 | 61 | 0 | 1 | 2 | 84 | 0 | 0 | 15
**consequence** | 1 | 23 | 10 | 0 | 8 | 12 | 5 | 9 | 59 | 0

Hence I adjusted the mapping based on the confusion matrix to below and got macro F1 of 0.64.


1: "concept", # "verification",
2: "consequence", # "disjunction",
3: "verification", # "concept",
5: "comparison", # "extent",
7: "example",
8: "extent", # "comparison",
9: "procedural", # "judgmental",
10: "cause",
11: "disjunction", # "consequence",
13: "judgmental", # "procedural"

I'm wondering if this should be the correct mapping? thanks!

**Comment by ShuyangCao** [comment]
Sorry for the late reply. The mapping I previously shared in this issue should be correct. I can have a check later.

**Comment by carriex** [comment]
Hi Shuyang, thanks for getting back to me! Please see attached the prediction I got from running the question type prediction model on the questions from `ontology_annotations.jsonl`. 
Columns: 
- question
- resolve_type (resolve_type from the annotation file)
- pred_type (question type predicted, without mapping)
- pred_type_old (question type using the mapping previously shared)
- pred_type_new (question type using the mapping from my last post)
- prob (model output)


The F1 of using old / new maps are 0.19 / 0.64 respectively. Could you help to check if you get the same pred_type from running the model? Thanks again for your help!

[question_with_ontology_annotation_with_type.csv](https://github.com/ShuyangCao/open-ended_question_ontology/files/7885941/question_with_ontology_annotation_with_type.csv)

**Comment by ShuyangCao** [comment]
I think I uploaded the wrong model 😅
Reuploaded and it should be correct now

**Comment by carriex** [comment]
The new one looks good now 👍 !
